### ISSUE TITLE
SFT Eval doesn't show any eval metrics/loss?

### ISSUE THREAD

**Issue by andrewsiah** [huggingface/trl#2203]
### System Info

colab notebook

### Information

- [ ] The official example scripts
- [X] My own modified scripts

### Tasks

- [X] An officially supported task in the `examples` folder
- [ ] My own task or dataset (give details below)

### Reproduction

https://colab.research.google.com/drive/1z31BEbar0wT4ONaiKsbnlHYzKjTV4WtV?usp=sharing

^ Notebook above is implementation.

```
from datasets import load_dataset
from trl import SFTConfig, SFTTrainer

num_train_data = 100

sft_config = SFTConfig(
    dataset_text_field="text",
    max_seq_length=512,
    output_dir="/tmp",
    num_train_epochs=4,
    learning_rate=1e-4,
    report_to='wandb',
    logging_steps=1,
    per_device_train_batch_size=8,
    per_device_eval_batch_size = 8
    # do_eval=True,
    # eval_on_start = True
)

trainer = SFTTrainer(
    model,
    train_dataset=formatted_dataset['train'].select(range(num_train_data)),
    eval_dataset=test_dataset['train'],
    args=sft_config,
    # formatting_func=formatting_prompts_func,
)

trainer.train()
```

### Expected behavior

There is no eval loss metrics? 

Only thing I get is the train loss metrics. (image below from output and wandb).

<img width="1550" alt="Screenshot 2024-10-08 at 9 41 55 PM" src="https://github.com/user-attachments/assets/1307847a-5880-45b5-b599-370aa0bd33ce">

<img width="1405" alt="Screenshot 2024-10-08 at 9 42 12 PM" src="https://github.com/user-attachments/assets/ea544656-006f-4780-b433-ba2ea3d8b8e0">

**Comment by qgallouedec** [comment]
You need to enable evaluation, see [the `transformers.TrainingArguments` doc](https://huggingface.co/docs/transformers/en/main_classes/trainer#transformers.TrainingArguments)

**Comment by RonanKMcGovern** [comment]
Yes, you have `do_eval` commented out.

Also, I believe you need:
```
        eval_strategy="steps",
        do_eval=True,
        eval_steps=0.1, #for eval every 10% of steps.
```

That said, I ran code today with those parameters and I'm not seeing an eval loss... (unless I manually add one to the trainer).

**Comment by qgallouedec** [comment]
```python
from trl import SFTConfig, SFTTrainer
from datasets import load_dataset

dataset = load_dataset("trl-lib/Capybara")

training_args = SFTConfig(output_dir="Qwen/Qwen2.5-0.5B-SFT", eval_steps=100, eval_strategy="steps", logging_steps=10)
trainer = SFTTrainer(
    args=training_args,
    model="Qwen/Qwen2.5-0.5B",
    train_dataset=dataset["train"],
    eval_dataset=dataset["test"],
)
trainer.train()
```

```
[2024-10-10 12:09:46,499] [INFO] [real_accelerator.py:203:get_accelerator] Setting ds_accelerator to cuda (auto detect)
wandb: WARNING The `run_name` is currently set to the same value as `TrainingArguments.output_dir`. If this was not intended, please specify a different run name by setting the `TrainingArguments.run_name` parameter.
wandb: Using wandb-core as the SDK backend. Please refer to https://wandb.me/wandb-core for more information.
wandb: Currently logged in as: qgallouedec (huggingface). Use `wandb login --relogin` to force relogin
wandb: Tracking run with wandb version 0.18.0
wandb: Run data is saved locally in /fsx/qgallouedec/trl/wandb/run-20241010_120948-fpkni0nj
wandb: Run `wandb offline` to turn off syncing.
wandb: Syncing run Qwen/Qwen2.5-0.5B-SFT
wandb: ⭐️ View project at https://wandb.ai/huggingface/huggingface
wandb: 🚀 View run at https://wandb.ai/huggingface/huggingface/runs/fpkni0nj
{'loss': 1.7838, 'grad_norm': 4.98770809173584, 'learning_rate': 4.991565452091768e-05, 'epoch': 0.01}                                                                                                                   
{'loss': 1.6715, 'grad_norm': 5.814300537109375, 'learning_rate': 4.983130904183536e-05, 'epoch': 0.01}                                                                                                                  
{'loss': 1.4957, 'grad_norm': 4.802212238311768, 'learning_rate': 4.974696356275304e-05, 'epoch': 0.02}                                                                                                                  
{'loss': 1.6878, 'grad_norm': 5.192160606384277, 'learning_rate': 4.966261808367072e-05, 'epoch': 0.02}                                                                                                                  
{'loss': 1.6875, 'grad_norm': 4.301702499389648, 'learning_rate': 4.95782726045884e-05, 'epoch': 0.03}                                                                                                                   
{'loss': 1.6357, 'grad_norm': 4.644138813018799, 'learning_rate': 4.9493927125506076e-05, 'epoch': 0.03}                                                                                                                 
{'loss': 1.6582, 'grad_norm': 4.883214473724365, 'learning_rate': 4.9409581646423755e-05, 'epoch': 0.04}                                                                                                                 
{'loss': 1.4887, 'grad_norm': 6.148635387420654, 'learning_rate': 4.9325236167341433e-05, 'epoch': 0.04}                                                                                                                 
{'loss': 1.5467, 'grad_norm': 4.263922214508057, 'learning_rate': 4.924089068825911e-05, 'epoch': 0.05}                                                                                                                  
{'loss': 1.6109, 'grad_norm': 3.8504934310913086, 'learning_rate': 4.915654520917679e-05, 'epoch': 0.05}                                                                                                                 
{'eval_loss': 1.6436591148376465, 'eval_runtime': 5.7794, 'eval_samples_per_second': 34.606, 'eval_steps_per_second': 4.326, 'epoch': 0.05}                                                                              
{'loss': 1.4516, 'grad_norm': 4.733430862426758, 'learning_rate': 4.907219973009447e-05, 'epoch': 0.06}                                                                                                                  
{'loss': 1.6327, 'grad_norm': 5.621842384338379, 'learning_rate': 4.898785425101215e-05, 'epoch': 0.06}                                                                                                                  
{'loss': 1.7306, 'grad_norm': 4.469166278839111, 'learning_rate': 4.890350877192983e-05, 'epoch': 0.07}                                                                                                                  
{'loss': 1.5902, 'grad_norm': 4.800514221191406, 'learning_rate': 4.881916329284751e-05, 'epoch': 0.07}                                                                                                                  
{'loss': 1.7242, 'grad_norm': 4.282039165496826, 'learning_rate': 4.8734817813765186e-05, 'epoch': 0.08} 
...
```

**Comment by gustavo1-articul8** [comment]
@qgallouedec , could you solve this?
